From c7ec63a2f7711bb4be26bf818ce07d5f8e0e08bf Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Fri, 1 Aug 2025 06:41:50 +0200
Subject: [PATCH 14/33] configure.ac: Make CJK fonts name configurable

Distributions might want to use a different CJK font family to support
Chinese, Japanese and Korean fonts. The MR introduces `--with-cjk-fonts`
configuration option which sets the fonts name.
---
 .gitignore                                        |  6 +++++-
 Makefile.am                                       |  4 +++-
 charset/{pdf.utf-8.heavy => pdf.utf-8.heavy.in}   |  2 +-
 charset/{pdf.utf-8.simple => pdf.utf-8.simple.in} |  2 +-
 configure.ac                                      | 15 +++++++++++++++
 5 files changed, 25 insertions(+), 4 deletions(-)
 rename charset/{pdf.utf-8.heavy => pdf.utf-8.heavy.in} (98%)
 rename charset/{pdf.utf-8.simple => pdf.utf-8.simple.in} (97%)

diff --git a/.gitignore b/.gitignore
index e256e473..638533ed 100644
--- a/.gitignore
+++ b/.gitignore
@@ -95,4 +95,8 @@ testfilters
 test-analyze
 Test_summary_final.txt
 test.pdf
-test.ps
\ No newline at end of file
+test.ps
+
+# ignore generated charset file
+charset/pdf.utf-8.simple
+charset/pdf.utf-8.heavy
diff --git a/Makefile.am b/Makefile.am
index ddebb5be..913457fd 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -21,7 +21,9 @@ EXTRA_DIST = \
 	$(doc_DATA) \
 	autogen.sh \
 	config.rpath \
-	libcupsfilters.pc.in
+	libcupsfilters.pc.in \
+	pdf.utf-8.heavy.in \
+	pdf.utf-8.simple.in
 
 EXTRA_DIST += \
 	data/makePDFfromPS.sh \
diff --git a/charset/pdf.utf-8.heavy b/charset/pdf.utf-8.heavy.in
similarity index 98%
rename from charset/pdf.utf-8.heavy
rename to charset/pdf.utf-8.heavy.in
index 6273d6c5..3f9021ad 100644
--- a/charset/pdf.utf-8.heavy
+++ b/charset/pdf.utf-8.heavy.in
@@ -37,5 +37,5 @@ charset utf8
 1E00 1EFF ltor single CourierNew CourierNew:bold CourierNew:italic CourierNew:bold:italic
 2000 21FF ltor single DejaVuSansMono DejaVuSansMono:bold DejaVuSansMono:oblique DejaVuSansMono:bold:oblique
 2200 23FF ltor single Symbol
-3000 9FFF ltor double ARPLUMingCN
+3000 9FFF ltor double @CJKFONTS@
 #0400 04FF ltor single FreeMono FreeMono:bold FreeMono:oblique FreeMono:bold:oblique
diff --git a/charset/pdf.utf-8.simple b/charset/pdf.utf-8.simple.in
similarity index 97%
rename from charset/pdf.utf-8.simple
rename to charset/pdf.utf-8.simple.in
index 42d1af0a..e49cbcf2 100644
--- a/charset/pdf.utf-8.simple
+++ b/charset/pdf.utf-8.simple.in
@@ -31,4 +31,4 @@ charset utf8
 0000 04FF ltor single monospace monospace:bold monospace:oblique monospace:bold:oblique
 0500 05FF rtol single monospace
 0600 06FF rtol single monospace
-3000 9FFF ltor double ARPLUmingCN
+3000 9FFF ltor double @CJKFONTS@
diff --git a/configure.ac b/configure.ac
index 4139012a..f68ba9e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -478,12 +478,26 @@ CXXFLAGS="$CXXFLAGS -D_GNU_SOURCE -DPOINTERHOLDER_TRANSITION=0"
 #                                      See /usr/include/qpdf/PointerHolder.hh
 CXXFLAGS="$CXXFLAGS -std=c++17"
 
+# =========
+# CJK FONTS
+# =========
+AC_ARG_WITH([cjk-fonts],
+	[AS_HELP_STRING([--with-cjk-fonts=value], [Set font name for CJK fonts (default: ARPLUmingCN).])],
+	[AS_IF([test "x$withval" != "x" -a "x$withval" != "xyes" -a "x$withval" != "xno"],
+	       [CJKFONTS="$withval"],
+	       [CJKFONTS="ARPLUmingCN"])],
+	[CJKFONTS="ARPLUmingCN"])
+
+AC_SUBST([CJKFONTS])
+
 # =====================
 # Prepare all .in files
 # =====================
 AC_CONFIG_FILES([
 	libcupsfilters.pc
 	Makefile
+	charset/pdf.utf-8.heavy
+	charset/pdf.utf-8.simple
 ])
 AC_OUTPUT
 
@@ -515,5 +529,6 @@ Build configuration:
 	dbus:            ${enable_dbus}
 	werror:          ${enable_werror}
 	test-font:       ${with_test_font_path}
+	cjk-fonts:       ${CJKFONTS}
 ==============================================================================
 ])
-- 
2.53.0

